include/nbl/builtin/hlsl/cpp_compat/matrix.h

```diff
@@ -0,0 +1,51 @@
+#ifndef _NBL_BUILTIN_HLSL_MATRIX_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATRIX_INCLUDED_
+
+#ifdef __cplusplus
+#include <glm/glm.hpp>
+#include <stdint.h>
+#endif
+
+namespace nbl::hlsl
+{
+
+#ifdef __cplusplus
+template<typename T, uint16_t N, uint16_t M>
+struct matrix final : private glm::mat<N,M,T>
+{
+    using Base = glm::mat<N,M,T>;
+    using Base::Base;
+    using Base::operator=;
+
+    template<uint16_t X, uint16_t Y> matrix(matrix<T, X, Y> const&) = delete;
+
+    matrix(matrix const&) = default;
+    matrix(Base const& base) : Base(base) {}
+
+    Base const& asBase() const { return *reinterpret_cast<const Base*>(this); }
+
+    
+    template<class T, uint16_t K>
+    friend matrix<T, N, K> operator*(matrix const& lhs, matrix<T, M, K> const& rhs){ return rhs.asBase() * lhs.asBase(); }
```

Comment:
HLSL matrices don't have an `operator*`, even HLSL202x matrices, IIRC?

Reply:
> Dxc on godbolt compiled * operators so hlsl2021 or 202x probably does

HLSL  `operator*` is only possible vs a scalar. 

`operator*` between matrices is misleading AF in HLSL is pointwise multiplication if I remember correctly, which is a mess.